packages/make-styles/src/types.ts

```diff
@@ -1,6 +1,8 @@
 import { Properties as CSSProperties } from 'csstype';
 
-export interface MakeStyles extends Omit<CSSProperties, 'animationName'> {
+export type MakeStylesCSSValue = string | 0;
+
+export interface MakeStyles extends Omit<CSSProperties<MakeStylesCSSValue>, 'animationName'> {
   // TODO Questionable: how else would users target their own children?
   [key: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Would it make sense to improve the typing of the `MakeStyles` in the prototype as it is supposed to be the only guard against the shorthands? Nested objects must have the same restrictions recursively.

Reply:
This is done in #20539 (giant PR 💥). I don't change typings in this PR as it only adds these functions.